### PR TITLE
Update docs for upcoming 0.2.0 release

### PR DESCRIPTION
**Pull request by mattdangerw**

### CODE REVIEW

**Comment by mattdangerw** [comment]
Also, note that this is currently cross linking the keras.io docs, so we should land this after we land the new keras.io section (or I can just ditch those links and submit them later).

**Review by chenmoneygithub — APPROVED** [review]
thanks for the documentation work!

File: API_DESIGN_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+class RougeL(keras.metrics.Metric):
+    def __init__(self):
+        if rouge_score is None:
+            raise ImportError(
+                'RougeL metrics requires the rouge_score package. '
+                '`pip install rouge-score`.')
+```
+
+## Keep computation inside TensorFlow graph
```
Comment (fchollet) [inline API_DESIGN_GUIDE.md:52]: Maybe worth pointing out the difference between "TF graph" (`tf.function`) and "XLA compilable" (`tf.function(jit_compile=True)`)

File: README.md [hunk 2]
```diff
@@ -4,78 +4,102 @@
...
+from tensorflow import keras
+
+# Tokenize some inputs with a binary label.
+vocab = ["[UNK]", "the", "qu", "##ick", "br", "##own", "fox", "jumped", "."]
+inputs = ["The quick brown fox jumped.", "The fox slept."]
+tokenizer = keras_nlp.tokenizers.WordPieceTokenizer(
+    vocabulary=vocab,
+    sequence_length=10,
+)
+X, Y = tokenizer(inputs), tf.constant([1, 0])
```
Comment (fchollet) [inline README.md:55]: Lowercase

File: STYLE_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
+# Style Guide
+
+## Use black
```
Comment (fchollet) [inline STYLE_GUIDE.md:3]: `black` (backticks)

File: STYLE_GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    >>> layer = keras_nlp.layers.Linear(4)
+    >>> layer(tf.zeros(8, 2)) == layer.b
+    True
+    """
+    def __init__(self, units=32, **kwargs):
+        super().__init__(**kwargs)
+        self.units = units
+
+    def build(self, input_shape):
+        super().build(input_shape)
```
Comment (fchollet) [inline STYLE_GUIDE.md:79]: Is the super call necessary?

File: STYLE_GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        self.w = self.add_weight(shape=(input_shape[-1], self.units))
+        self.b = self.add_weight(shape=(self.units,))
+
+    def call(self, inputs):
```
Comment (fchollet) [inline STYLE_GUIDE.md:83]: Might be worth documenting `mask` and possibly `training`

File: ROADMAP.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+  us to focus on usability and API standardization, and produce objects that
+  have a longer lifespan than the average research project.
+
+- **KerasNLP is not a repository of blackbox end-to-end solutions.**
+  KerasNLP is focused on modular and reusable building blocks. In the process
+  of developing these building blocks, we will by necessity implement
+  end-to-end workflows, but they're intended purely for demonstration and
+  grounding purposes, they're not our main deliverable.
+
+- **KerasNLP is not a repository of low-level string ops, like tf.text.**
```
Comment (fchollet) [inline ROADMAP.md:41]: Backticks

File: ROADMAP.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+- Pre-trained sub-word tokenizers for any language. Training a tokenizer can
+  add a lot of friction to a project, particularly when working working in a
+  language where examples are less readily available. We would like to support
+  a pre-trained tokenization offering that allows a user to choose a tokenizer,
+  language, and vocabulary size and then download an off the shelf vocabulary.
+- A standardized way to training tokenizer vocabularies. As another way to
+  reduce the friction of training a tokenizer, we should offer a standardized
+  experience for training new vocabularies.
+
+### Scalable examples of popular model architectures using KerasNLP
```
Comment (fchollet) [inline ROADMAP.md:85]: "scalable" -> should we mention DTensor support somewhere? This will be important for model parallelism.

File: ROADMAP.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+
+At this moment in time, we have no set citation bar for development, but due to
+the newness of the library we want to focus our efforts on a small subset of the
+best known and most effective NLP techniques.
+
+Proposed components should usually either be part of a very well known
+architecture or contribute in some meaningful way to the usability of an
+end-to-end workflow.
+
+## Pre-trained modeling workflows
```
Comment (fchollet) [inline ROADMAP.md:158]: Pretrained/pretraining

File: ROADMAP.md [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+
+Pre-training many modern NLP models is prohibitively expensive and
+time-consuming for an average user. A key goal with for the KerasNLP project is
+to have KerasNLP components available in a pre-trained model offering of some
+form.
+
+We are working with the rest of the Tensorflow ecosystem (e.g. TF Hub,
```
Comment (fchollet) [inline ROADMAP.md:165]: TensorFlow

(also we aren't exactly working with TF Hub, are we?)

**Review by fchollet — APPROVED** [review]
LGTM, thank you!

File: STYLE_GUIDE.md (hunk 5 above)
Comment (mattdangerw) [reply STYLE_GUIDE.md:113]: This is the style guide, so will keep discussion of how to actually handle masking out of here. But added a not that masking behavior of the layer should be documented in the class level docstring.

**Comment by mattdangerw** [comment]
Submitting this without the keras.io links before I cut 0.2.0. I will push the keras.io links back as soon as the section is live.
